### PR TITLE
ff2mpv: init at 4.0.0

### PR DESCRIPTION
**Pull request by Enzime**
###### Description of changes

The native client for https://addons.mozilla.org/en-US/firefox/addon/ff2mpv/

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by QuantMint** [comment]
Result of `nixpkgs-review pr 166581` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ff2mpv</li>
  </ul>
</details>

**Comment by QuantMint** [comment]
How is this supposed to work? I installed the package and the extension but mpv never opens.

**Comment by Enzime** [comment]
`$out/lib/mozilla/native-messaging-hosts/ff2mpv.json` needs to be placed in `~/.mozilla/native-messaging-hosts/` which I do in my `home.nix` with:

```nix
{
  home.file.".mozilla/native-messaging-hosts/ff2mpv.json".source = "${pkgs.ff2mpv}/lib/mozilla/native-messaging-hosts/ff2mpv.json";
}
```

**Comment by Enzime** [comment]
It seems like you can also do this with Firefox:

```nix
programs.firefox.override {
  extraNativeMessagingHosts = [ pkgs.ff2mpv ];
}
```

**Review by Tungsten842 — APPROVED** [review]
Works well

**Comment by gador** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 166581` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>ff2mpv</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
ff2mpv:
<details>

got build log for '/nix/store/vjrdifwdwcmkk4g5hvh1pjzss4y23amp-ff2mpv-4.0.0' from 'daemon'
When evaluating attribute ‘ff2mpv’:
warning: patch-phase
`patchPhase` should not be overridden, use `postPatch` instead.

Near pkgs/applications/misc/ff2mpv/default.nix:16:3:

```
   |
16 |   patchPhase = ''
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/patch-phase.md
</details>
</details>

File: pkgs/applications/misc/ff2mpv/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  src = fetchFromGitHub {
+    owner = "woodruffw";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-sxUp/JlmnYW2sPDpIO2/q40cVJBVDveJvbQMT70yjP4=";
+  };
+
+  buildInputs = [ python3 mpv ];
+
+  patchPhase = ''
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/ff2mpv/default.nix:16]: Please use postPatch.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/ff2mpv/default.nix (hunk 1 above)
Comment (Enzime) [reply pkgs/applications/misc/ff2mpv/default.nix]: Fixed 👍

**Comment by bbigras** [comment]
> You can also do this directly with the Firefox wrapper in Nixpkgs:
> 
> ```nix
> programs.firefox.override {
>   extraNativeMessagingHosts = [ pkgs.ff2mpv ];
> }
> ```

Any ideas why it wouldn't work with home-manager's `programs.firefox`?

```nix
{
  programs.firefox = {
    package = (pkgs.firefox-bin.override {
      extraNativeMessagingHosts = [ pkgs.ff2mpv ];
    });
  };
}
```

Setting `home.file.".mozilla/native-messaging-hosts/ff2mpv.json"` seems to work.

File: pkgs/applications/misc/ff2mpv/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    owner = "woodruffw";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-sxUp/JlmnYW2sPDpIO2/q40cVJBVDveJvbQMT70yjP4=";
+  };
+
+  buildInputs = [ python3 mpv ];
+
+  postPatch = ''
+    patchShebangs .
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/ff2mpv/default.nix:17]: Can we be more specific which files need patching?

**Pull request merged**